v_0_1_0/master_template.go

```diff
@@ -10,6 +10,94 @@ users:
        - "{{ $user.PublicKey }}"
 {{end}}
```

Comment:
Should it go to the v_0_1_0? That would mean we have inconsistent cluster version again. WDYT?

Reply:
i can not answer this. May be @xh3b4sd ? from what i see we are not using versioninig yet

Reply:
I think we should start version our own code better. When a new feature is added lets have a new folder.

Reply:
We said we hold off on versioning until we have a agreed upon approach to it and can easily change/use it. Just adding versions in k8scc now would again break stuff I fear (e.g. operators have fixed version in code currently)

Reply:
Do we have a code that already use this v_X_Y_Z folders?

Thanks @kopiczko for raising this, but i think we should continue this discussion out of this PR.

Reply:
I do not agree that "overdoing" versioning breaks something. The opposite is the case. We potentially just break stuff if we throw everything into the same branch of our code. Creating a new version folder here is more work but also makes us starting to gain experiences on how to actually version our software. There will not be the final solution that makes everything easier. We have to do it. We have to learn from it. There is no discussion necessary that we will have to copy code into versioned folders. We have to do it. So why not start now? How the folders are named is TBD though. But that we have to do it is not questionable AFAIK.

Reply:
I would like to have full support in the operators for this (i.e. more than one folder) before we just add a folder. Full support means we just can choose based on TPR or whatever and have no hardcoded thing in the code that chooses a version. I am not a big fan of changing stuff in dependencies without having the support for it in the other components ready, especially if it's such central components like k8scc and cluster operators where we are rolling out new version a lot recently. These changes are hard breaks and they block others from getting on with their issues that they have priority on.
The case where we break stuff hasn't happened a lot lately, since we do actually test on gauss (and hopefully on lycan) when we change things. I don't see versioning helping a lot there TBH, QA and testing will and those can be done based on branches (and other mechanics) already without versioning.

Reply:
Fully agree with you that "only we are in charge of our own destinies".

I just don't know what and how treats this version folder. Let's may be start with that. If you can point me to the right code, then next time i'll try to bump the version. Or if there is no such code, then it may be good that we started this discussion and it's time to implement one or we even don't need one at all. And these are only my assumptions. But i know nothing about versioning, sorry.